mernarezk/assignment2/TreeNode.java

```diff
@@ -0,0 +1,20 @@
+package Assignment2;
+
+/**
+ * simple TreeNode class that takes only int values for simplicity and as both
+ * problems defined for int values.
+ * 
+ * @author Merna
+ *
+ */
+public class TreeNode {

```

Comment:
For easier way to find the ancestors it is okay to hold a reference to the parent of the node.

Reply:
Okay I added to my TreeNode structure a parent node.